### PR TITLE
Simplify GitHub repo connect flow

### PR DESCRIPTION
**Pull request by tonyjhuang**
### Description

Removes the "configure existing connection" choice from the repo picker. 

Also aims to simplify the top level `linkGitHubRepository()` by adding some helper methods

### CODE REVIEW

File: src/init/features/apphosting/repo.ts [hunk 1]
```diff
@@ -87,84 +91,111 @@ export async function linkGitHubRepository(
...
-    let refreshedConn = conn;
-    while (refreshedConn.installationState.stage !== "COMPLETE") {
-      refreshedConn = await promptAppInstall(conn);
-    }
-    existingConns.push(refreshedConn);
+
+  if (existingConns.length === 0) {
+    await ensureSecretManagerAdminGrant(projectId);
+    existingConns.push(
+      await ensureFullyInstalledConnection(projectId, location, generateConnectionId(), oauthConn),
```
Comment (mathu97) [inline src/init/features/apphosting/repo.ts:101]: NIT: the naming here doesn't make me think that a new connection would be created.

File: src/init/features/apphosting/repo.ts [hunk 2]
```diff
@@ -87,84 +91,111 @@ export async function linkGitHubRepository(
...
+    );
   }
 
-  let { remoteUri, connection } = await promptRepositoryUri(projectId, existingConns);
-  while (CONFIGURE_REMOTE_URI_CHOICES.includes(remoteUri)) {
-    if (remoteUri === ADD_REPO_CHOICE) {
```
Comment (mathu97) [inline src/init/features/apphosting/repo.ts:115]: why remove this? I like that it makes the UX simpler but would this not mean that we would be creating a new Connection whenever the user wants to make a change to the repos available to the Cloud Build Github app?

Though i'm not too sure about the consequences of having many connections ... are there any impact on billing?

**Review by mathu97 — COMMENTED** [review]
LGTM as long as this was tested manually. There doesn't seem to be any logic changes - mostly refactoring.

File: src/init/features/apphosting/repo.ts (hunk 1 above)
Comment (tonyjhuang) [reply src/init/features/apphosting/repo.ts:101]: Updated to `createFullyInstalledConnection()`

File: src/init/features/apphosting/repo.ts (hunk 2 above)
Comment (tonyjhuang) [reply src/init/features/apphosting/repo.ts:115]: GCB `connection` resources are free so it doesn't hurt to create multiple. The idea is that once we have a Firebase GitHub app, we can dedupe `connections` on behalf of the user in our redirect handler.

**Review by mathu97 — APPROVED** [review]

**Pull request merged**